### ISSUE TITLE
memory  leak in sample

### ISSUE THREAD

**Issue by fatlinlin** [asus4/tf-lite-unity-sample#145]
**Environment (please complete the following information):**
 - OS/OS Version:  iOS 13
 - Source Version: master
 - Unity Version:  unity2019.4.24

**Describe the bug**
memory  leak in sample

**To Reproduce**
1. add button to Pose Scene (to navigate back to Index Scene)
2. run the sample on iphone
3. navigate scene from Index to pose scene many times
4. find memory increase by xcode

**Comment by asus4** [comment]
Thank you for reporting. will check this later.

**Comment by fatlinlin** [comment]
> Thank you for reporting. will check this later.

hi, I find  memory leak is caused by tensorflow lite interpreter . Is there any way to release the these memory.

**Comment by fatlinlin** [comment]
![image](https://user-images.githubusercontent.com/3388306/126299700-0737ea4f-5929-4143-9a78-a917383de22f.png)

**Comment by mgrman** [comment]
@asus4 Seem allocating GCHandle modelDataHandle in the Interpreter.cs file and never freeing it can be the cause. The byte array for the model is pinned in memory but never freed, therefore the model array can never be collected from memory. Calling modelDataHandle.Free() in the Dispose method fixes this potential leak.

**Comment by asus4** [comment]
Thanks @mgrman 
Fixed in #174

**Comment by mgrman** [comment]
@asus4 thanks for the quick integration. But seems there are more GCHandles there. Don't they need to be freed as well?

**Comment by asus4** [comment]
Thanks @mgrman, fixed it.

Probably this should be reported as an issue in the official [TensorFlowLitePlugin](https://github.com/tensorflow/tensorflow/blob/f959fe4761b7ea3b3e474652e04799e9f6e09f06/tensorflow/lite/experimental/examples/unity/TensorFlowLitePlugin/Assets/TensorFlowLite/SDK/Scripts/Interpreter.cs) as well.

**Comment by sonnybsj** [comment]
@asus4 Regarding [this commit](https://github.com/asus4/tf-lite-unity-sample/pull/174/commits/f11d234bbb04e85c2664560acfb81f86273769ef) and this line:

https://github.com/asus4/tf-lite-unity-sample/blob/8b9a2381fb8c039a2918ab7a7efc7499cb86fb07/Packages/com.github.asus4.tflite/Runtime/Interpreter.cs#L61

Wouldn't it be safer to call `GCHandle.Free()` in `Dispose()` instead of right after the call to `TfLiteModelCreate()`?

In one of my custom scenes, my model stopped working due to this change when I built to `iOS` (even though it was fine in-editor on `macOS`).

Note that I didn't see a problem with any of the project sample scenes, so it might be an issue on my end... But just thought I'd note it here.

**Comment by asus4** [comment]
Thanks @sonnybsj

I've modified it to call `GCHandle.Free` in `Dispose()`. Could you try to see if this solves your crashes?

**Comment by sonnybsj** [comment]
@asus4 yes, it solves my crashes.

Note that for me, moving `modelDataHandle.Free()` to `Dispose()` was enough to fix the crashes. But the changes you made to free `inputDataHandles` and 'outputDataHandles` in `Dispose()` looks good.

Also, note that it wasn't necessarily a crash. I had behaviour where the MoveNet skeleton tracking wasn't working once I made a build.

I noticed you flagged the issue in #180, but I think #177 #181 #182 could be related as well.
